Guide/src/reference/backends/storage/nvme.md

```diff
@@ -0,0 +1,59 @@
+# NVMe Storage Backend
+
+## Overview
+
+OpenHCL includes an NVMe driver that enables it to interact with NVMe storage devices assigned to VTL2. This is particularly important in Azure Boost environments where storage is exposed as NVMe devices.
+
+The NVMe driver in Underhill (OpenHCL's userspace component) provides a safe, Rust-based implementation for managing NVMe storage devices through VFIO (Virtual Function I/O).
+
+## Key Components
+
+### NVMe Driver
+
+The NVMe driver (`nvme_driver` crate) provides the core functionality for interacting with NVMe devices:
+
+- **Device Initialization**: Handles NVMe controller initialization, queue setup, and namespace discovery
+- **I/O Operations**: Provides asynchronous read/write operations through submission and completion queues
+- **Save/Restore Support**: Enables device state preservation during servicing operations (when supported)
+
+For detailed implementation information, see the [NVMe Driver API documentation](https://openvmm.dev/rustdoc/linux/nvme_driver/index.html).
+
+### NVMe Manager
+
+The NVMe manager (`underhill_core::nvme_manager`) coordinates multiple NVMe devices and provides:
+
+- **Device Registry**: Tracks and manages multiple NVMe devices by PCI ID
+- **Multi-threaded Architecture**: Uses mesh RPC for concurrent cross-device operations while serializing per-device requests
+- **Namespace Resolution**: Resolves disk configurations to specific NVMe namespaces
+- **Lifecycle Management**: Handles device initialization, operation, and shutdown
+
+### DMA Manager Integration
+
+NVMe devices work closely with the [DMA Manager](../../openhcl/dma_manager.md) for memory management:
+
+- **Private Pool**: When available, NVMe devices use the private pool for persistent DMA allocations that survive servicing
+- **VTL Permissions**: The DMA manager ensures proper VTL0 access permissions for memory buffers
+- **Save/Restore Support**: Persistent allocations enable NVMe keepalive during servicing operations
```

Comment:
Don't need to include this, because these are all simple intrinsics of the dma manager.